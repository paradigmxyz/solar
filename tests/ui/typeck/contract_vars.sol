function f() {}
event E1();
error E2();

contract C {
    f a; //~ ERROR: name has to refer to a valid user-defined type
    E1 b; //~ ERROR: name has to refer to a valid user-defined type
    E2 c; //~ ERROR: name has to refer to a valid user-defined type
}
