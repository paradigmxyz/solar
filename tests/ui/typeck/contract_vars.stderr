error: name has to refer to a valid user-defined type
  --> ROOT/tests/ui/typeck/contract_vars.sol:LL:CC
   |
LL |     f a;
   |     ^
   |

error: name has to refer to a valid user-defined type
  --> ROOT/tests/ui/typeck/contract_vars.sol:LL:CC
   |
LL |     E1 b;
   |     ^^
   |

error: name has to refer to a valid user-defined type
  --> ROOT/tests/ui/typeck/contract_vars.sol:LL:CC
   |
LL |     E2 c;
   |     ^^
   |

error: aborting due to 3 previous errors

